packages/ckeditor5-image/src/imageupload/imageuploadui.js

```diff
@@ -8,56 +8,155 @@
 
-		// Setup `imageUpload` button.
 		editor.ui.componentFactory.add( 'imageUpload', locale => {
-			const view = new FileDialogButtonView( locale );
-			const command = editor.commands.get( 'imageUpload' );
-			const imageTypes = editor.config.get( 'image.upload.types' );
-			const imageTypesRegExp = createImageTypeRegExp( imageTypes );
+			const imageUploadView = new ImageUploadPanelView( locale, prepareIntegrations( editor ) );
+
+			const dropdownView = imageUploadView.dropdownView;
+			const panelView = dropdownView.panelView;
+			const splitButtonView = dropdownView.buttonView;
 
-			view.set( {
-				acceptedType: imageTypes.map( type => `image/${ type }` ).join( ',' ),
-				allowMultipleFiles: true
-			} );
+			splitButtonView.actionView = this._createFileDialogButtonView( locale );
 
-			view.buttonView.set( {
-				label: t( 'Insert image' ),
-				icon: imageIcon,
-				tooltip: true
-			} );
+			panelView.children.add( imageUploadView );
 
-			view.buttonView.bind( 'isEnabled' ).to( command );
+			return this._setUpDropdown( dropdownView, imageUploadView, command );
+		} );
+	}
+
+	/**
+	 * Sets up the dropdown view.
+	 *
+	 * @param {module:ui/dropdown/dropdownview~DropdownView} dropdownView A dropdownView.
+	 * @param {module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView} imageUploadView An imageUploadView.
+	 * @param {module:core/command~Command} command An imageUpload command
+	 *
+	 * @private
+	 * @returns {module:ui/dropdown/dropdownview~DropdownView}
+	 */
+	_setUpDropdown( dropdownView, imageUploadView, command ) {
+		const editor = this.editor;
+		const t = editor.t;
+		const insertButtonView = imageUploadView.insertButtonView;
+
+		dropdownView.bind( 'isEnabled' ).to( command );
 
-			view.on( 'done', ( evt, files ) => {
-				const imagesToUpload = Array.from( files ).filter( file => imageTypesRegExp.test( file.type ) );
+		dropdownView.on( 'change:isOpen', () => {
+			const selectedElement = editor.model.document.selection.getSelectedElement();
 
-				if ( imagesToUpload.length ) {
-					editor.execute( 'imageUpload', { file: imagesToUpload } );
+			if ( dropdownView.isOpen ) {
+				imageUploadView.focus();
+
+				if ( isImage( selectedElement ) ) {
+					imageUploadView.imageURLInputValue = selectedElement.getAttribute( 'src' );
+					insertButtonView.label = t( 'Update' );
+				} else {
+					imageUploadView.imageURLInputValue = '';
+					insertButtonView.label = t( 'Insert' );
 				}
-			} );
+			}
+		} );
 
-			return view;
+		imageUploadView.delegate( 'submit', 'cancel' ).to( dropdownView );
+		this.delegate( 'cancel' ).to( dropdownView );
+
+		dropdownView.on( 'submit', () => {
+			closePanel();
+			onSubmit();
+		} );
+
+		dropdownView.on( 'cancel', () => {
+			closePanel();
+		} );
+
+		function onSubmit() {
+			const selectedElement = editor.model.document.selection.getSelectedElement();
+
+			if ( selectedElement && isImage( selectedElement ) ) {
```

Comment:
```suggestion
			if ( isImage( selectedElement ) ) {
```

[`isImage()` checks whether passed an element.](https://github.com/ckeditor/ckeditor5/blob/38aad2abe9a85061f481028eb1d68a0ac2b4a6fe/packages/ckeditor5-image/src/image/utils.js#L41-L43)